### PR TITLE
Spec for the new Query Role API

### PR DESCRIPTION
**Pull request by albertzaharovits**
The rest specs for the new query role API in 8.15 https://github.com/elastic/elasticsearch/pull/108733 .
The docs for it are at https://github.com/elastic/elasticsearch/pull/110473 .

### CODE REVIEW

File: specification/security/_types/RoleDescriptor.ts [hunk 1]
```diff
@@ -52,6 +52,7 @@ export class RoleDescriptor {
    * @doc_id run-as-privilege
    */
   run_as?: string[]
+  description?: string
```
Comment (albertzaharovits) [inline specification/security/_types/RoleDescriptor.ts:55]: This is not related to the new API.
Role Descriptors have a description field since https://github.com/elastic/elasticsearch/pull/107088 .

File: specification/security/query_roles/QueryRolesRequest.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+import { RoleAggregationContainer } from './types'
+import { RequestBase } from '@_types/Base'
+import { integer } from '@_types/Numeric'
+import { Sort, SortResults } from '@_types/sort'
+
+/**
+ * Retrieves roles in a paginated manner. You can optionally filter the results with a query.
+ * @rest_spec_name security.query_role
```
Comment (albertzaharovits) [inline specification/security/query_roles/QueryRolesRequest.ts:27]: Note this is singular here `query_role`, but the specs dir is plural (`query_roles`).
ES endpoints are singular, eg https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-query-api-key.html , but I see it in the specs that the dir is named `query_api_keys`.

**Comment by l-trotta** [comment]
is there no json spec for this in the server code? we cannot validate this nor generate the output without that at the moment

File: specification/security/query_roles/QueryRolesRequest.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { RoleAggregationContainer } from './types'
```
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply specification/security/query_roles/QueryRolesRequest.ts:20]: Yes, good catch!

**Comment by albertzaharovits** [comment]
Thanks @l-trotta . I've addressed your review. Please take another look.

**Comment by l-trotta** [comment]
following what the [validator](https://github.com/elastic/elasticsearch-specification/actions/runs/9842949495/job/27173046418?pr=2676) is saying, `description` should also be added as property to `RoleDescriptorRead`

**Review by l-trotta — APPROVED** [review]

**Comment by albertzaharovits** [comment]
Thank you for your help on this @l-trotta ! Much appreciated.
